modules/services/gpg-agent.nix

```
@@ -237,23 +257,25 @@ in {
           "pinentry-program ${pkgs.pinentry.${cfg.pinentryFlavor}}/bin/pinentry"
           ++ [ cfg.extraConfig ]);
 
-      home.sessionVariablesExtra = optionalString cfg.enableSshSupport ''
-        if [[ -z "$SSH_AUTH_SOCK" ]]; then
-          export SSH_AUTH_SOCK="$(${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket)"
-        fi
-      '';
+      home.sessionVariables = optionalAttrs cfg.enableSshSupport {
+        SSH_AUTH_SOCK = "${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket";
+      };
```

this didn't work for me with `nushell`. May I ask if you have a better idea how to fix this? @lheckemann

---

I don't think this will work even with bash, since this will result in the variable being set to the command string rather than the command's output (which is what the `$()` in the previous code was for, like nu's `()`).

---

```suggestion
      home.sessionVariables = optionalAttrs cfg.enableSshSupport {
        SSH_AUTH_SOCK = "$XDG_RUNTIME_DIR/gnupg/S.gpg-agent.ssh";
      };
```
this is what I meant in my comment in the old PR, and that should avoid the need for any nushell-specific code. I'm just not sure if there are circumstances in which this value will be incorrect.

---

hm... I don't think that this will work because in nu, if you want to access an environment-veriable in nushell, you'll have to write `$env.<env-variable>`.

---

You're right! I thought this used some shell-agnostic mechanism like pam, but it turns out I was wrong :(

So let's get rid of this part of the change completely, sorry for leading you down the wrong path there!

---

> So let's get rid of this part of the change completely

hm... so you mean that I should remove the `home.sessionVariables` entry completely?

> sorry for leading you down the wrong path there!

no need to apologize :)

---

Yeah basically just bring back what it had originally.